searchlib/src/main/java/com/yahoo/searchlib/rankingexpression/rule/ReferenceNode.java

```diff
@@ -79,7 +79,7 @@ public StringBuilder toString(StringBuilder string, SerializationContext context
                 path = new ArrayDeque<>();
             String myPath = getName() + getArguments().expressions();
             if (path.contains(myPath))
-                throw new IllegalStateException("Cycle in ranking expression function: " + path);
+                throw new IllegalStateException("Cycle in ranking expression function '" + getName() + "' called from: " + path);
```

Comment:
Should this be IllegalArgumentException too, it is error in the input ?

Reply:
Yes.